Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -498,11 +506,13 @@ Bitmap::~Bitmap()
     delete[] m_palette;
 }
 
-void Bitmap::set_mmap_name([[maybe_unused]] DeprecatedString const& name)
+void Bitmap::set_mmap_name([[maybe_unused]] String const& name)
 {
     VERIFY(m_needs_munmap);
 #ifdef AK_OS_SERENITY
-    ::set_mmap_name(m_data, size_in_bytes(), name.characters());
+    const ByteBuffer name_bytes_null_terminated = null_terminated_string(name);
```

Comment:
This is quite a strange set of changes. The mmap syscall in Serenity (where we actually use this name) should have no requirement to be null terminated. If it currently does, this is a defect in our Kernel. None of the other syscalls that have a char const* as input require that, we take a `Syscall::StringArgument` instead, which is a ptr + size.

A better change would be to update the syscall in the kernel to take a StringArgument, and just pass the formatted name in via the String::view() or other observer APIs.

Reply:
I checked and the syscall to uses `StringArgument`s
The c-like API just uses `{name, strlen(name)}`
I proposed adding a LibCore version that takes a propper `StringView`, à la `Core::System::mmap`